lib/flight_scheduler/configuration/loader.rb

```diff
@@ -0,0 +1,85 @@
+#==============================================================================
+# Copyright (C) 2020-present Alces Flight Ltd.
+#
+# This file is part of FlightSchedulerController.
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which is available at
+# <https://www.eclipse.org/legal/epl-2.0>, or alternative license
+# terms made available by Alces Flight Ltd - please direct inquiries
+# about licensing to licensing@alces-flight.com.
+#
+# FlightSchedulerController is distributed in the hope that it will be useful, but
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, EITHER EXPRESS OR
+# IMPLIED INCLUDING, WITHOUT LIMITATION, ANY WARRANTIES OR CONDITIONS
+# OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY OR FITNESS FOR A
+# PARTICULAR PURPOSE. See the Eclipse Public License 2.0 for more
+# details.
+#
+# You should have received a copy of the Eclipse Public License 2.0
+# along with FlightSchedulerController. If not, see:
+#
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on FlightSchedulerController, please visit:
+# https://github.com/openflighthpc/flight-scheduler-controller
+#==============================================================================
+require 'active_support/core_ext/hash/keys'
```

Comment:
Partially requiring `active_support` is fine, however can it be done in a single location? Otherwise it'll slowly get required piecemeal all over the place.

By moving it to an initializer, it's allow easier to track which bits are in use.

Reply:
The advantage of co-locating the `require` with the code that requires it is that its easy to see which classes require which parts of active support.

With co-located `require`s we can determine which parts of active support are `require`d by running `git grep \brequire\b.*active_support`.

If we move all `require`s of active support to an initializer file it becomes hard to determine which classes require which parts of active support; and whether they still require those parts or not.